packages/material-ui-based-components/src/components/Accordion.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useContext } from 'react';
+import { getRenderer, FormContext } from '@aemforms/af-react-renderer';
+import { withRuleEnginePanel } from '../shared/withRuleEngine';
+import { PROPS_PANEL } from '../utils/types';
+
+import Accordion from '@mui/material/Accordion';
+import AccordionSummary from '@mui/material/AccordionSummary';
+import AccordionDetails from '@mui/material/AccordionDetails';
+import Typography from '@mui/material/Typography';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+
+const AccordionComponent = function (props: PROPS_PANEL) {
+    const [expanded, setExpanded] = React.useState<number>(0);
+    // @ts-ignore
+    const mappings = useContext(FormContext).mappings;
+    const items = props.items || [];
```

Comment:
Hidden items should not render here. Please filter props.item with the visible property.

Reply:
Checked as items= props.visible?props.items:[];